File: php/change_index_settings.php
<?php

# Install the API client: https://www.algolia.com/doc/api-client/getting-started/install/php/?client=php
require __DIR__.'/vendor/autoload.php';

use Algolia\AlgoliaSearch\Api\SearchClient;

$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->load();

# Get your Algolia Application ID and (admin) API key from the dashboard: https://www.algolia.com/account/api-keys
# and choose a name for your index. Add these environment variables to a `.env` file:
$ALGOLIA_APP_ID = $_ENV['ALGOLIA_APP_ID'];
$ALGOLIA_API_KEY = $_ENV['ALGOLIA_API_KEY'];
$ALGOLIA_INDEX_NAME = $_ENV['ALGOLIA_INDEX_NAME'];

# Start the API client
# https://www.algolia.com/doc/api-client/getting-started/instantiate-client-index/
$client = SearchClient::create($ALGOLIA_APP_ID, $ALGOLIA_API_KEY);

# Set index settings
# https://www.algolia.com/doc/api-reference/api-methods/set-settings/
$res = $client->setSettings(
    $ALGOLIA_INDEX_NAME, 
    [
      'searchableAttributes' => ['actors', 'genre'],
      'customRanking' => ['desc(rating)'],
    ],
    // Option to forward the same settings to the replica indices, when forwarding settings, please make sure your replicas already exist.
    [
      'forwardToReplicas' => true
    ]
    );

$client->waitForTask($ALGOLIA_INDEX_NAME, $res['taskID']);

# Printing settings 
# https://www.algolia.com/doc/api-reference/api-methods/get-settings/
print("Index settings:\n");
$settings = $client->getSettings($ALGOLIA_INDEX_NAME, 1);
var_dump($settings);

?>

File: go/change_index_settings.go
package main

import (
	"fmt"
	"log"
	"os"
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
	"github.com/joho/godotenv"
)

func PrintErrAndExit(err error) {
	fmt.Println(err)
	os.Exit(1)
}

func main() {
	err := godotenv.Load()
	if err != nil {
		log.Fatal("Error loading .env file")
	}
	// Algolia client credentials
	appID, apiKey, indexName := os.Getenv("ALGOLIA_APP_ID"), os.Getenv("ALGOLIA_API_KEY"), os.Getenv("ALGOLIA_INDEX_NAME")

	// Start the API client
	// https://www.algolia.com/doc/libraries/sdk/methods/search#go
	client, err := search.NewClient(appID, apiKey)
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		fmt.Printf("Client error")
		panic(err)
	}

	//Setting settings
	//https://www.algolia.com/doc/libraries/sdk/methods/search/set-settings
	resSetSettings, err := client.SetSettings(client.NewApiSetSettingsRequest(
		indexName,
		search.NewEmptyIndexSettings().SetSearchableAttributes(
            []string{"actors", "genre"},
        )).WithForwardToReplicas(true),
	)
	if err != nil {
		PrintErrAndExit(err)
	}
	_, err = client.WaitForTask(indexName, resSetSettings.TaskID)
	if err != nil {
		PrintErrAndExit(err)
	}

	//Printing settings
	//https://www.algolia.com/doc/api-reference/api-methods/get-settings/
	fmt.Println("Index settings:")
	resGetSettings, err := client.GetSettings(client.NewApiGetSettingsRequest(
	indexName).WithGetVersion(2))
		if err != nil {
		// handle the eventual error
		panic(err)
	}

	fmt.Println(resGetSettings)
}